xds/internal/resolver/serviceconfig.go

```diff
@@ -324,3 +370,33 @@ func (il *interceptorList) NewStream(ctx context.Context, ri iresolver.RPCInfo,
 	}
 	return newStream(ctx, func() {})
 }
+
+type clusterKey struct{}
+
+func GetPickedCluster(ctx context.Context) string {
+	return ctx.Value(clusterKey{}).(clusterInfoForLB).clusterName
+}
+
+func GetRequestHash(ctx context.Context) uint64 {
+	return ctx.Value(clusterKey{}).(clusterInfoForLB).requestHash
+}
+
+// GetPickedClusterForTesting returns the cluster in the context; to be used
+// for testing only.
+func GetPickedClusterForTesting(ctx context.Context) string {
+	return GetPickedCluster(ctx)
+}
+
+type clusterInfoForLB struct {
+	clusterName string
+	requestHash uint64
+}
+
+// SetPickedClusterAndRequestHash adds the selected cluster to the context for the
```

Comment:
Comment, it not only cluster, also request hash.

Reply:
Done.